src/index.ts

```diff
@@ -1,90 +1,122 @@
 /**
  * Helper for async/await error handling. Resolves a promise and passes an error if one exists. Promises of any type with any return value are allowed.
- * @param promise Function or promise
+ * @param promise type of promise or a function that returns a promise or an object that contians either a promise or a function that returns a promise
```

Comment:
Change this to: "Promise or a function that returns a promise or an object that contains either a promise or a function that returns a promise"

Reply:
Done 👍